editor/src/components/canvas/controls/grid-controls.tsx

```diff
@@ -412,13 +406,27 @@ export const GridControls = controlForStrategyMemoized(() => {
   }, [grids, jsxMetadata])
 
   const shadow = React.useMemo(() => {
-    return cells.find((cell) => EP.toUid(cell.elementPath) === dragging)
+    return cells.find((cell) => EP.toUid(cell.elementPath) === dragging) ?? null
   }, [cells, dragging])
 
   const [initialShadowFrame, setInitialShadowFrame] = React.useState<CanvasRectangle | null>(
     shadow?.globalFrame ?? null,
   )
 
+  const gridPath = React.useMemo(() => {
+    if (shadow == null) {
+      return null
+    }
+    return EP.parentPath(shadow.elementPath)
+  }, [shadow])
+
+  useSnapAnimation({
+    targetRootCell: targetRootCell,
+    controls: controls,
+    shadowFrame: initialShadowFrame,
+    gridPath: gridPath,
```

Comment:
does useSnapAnimation depend on reference equality here? if not, then there's no point in useMemo'ing gridPath

Reply:
https://github.com/concrete-utopia/utopia/pull/6046/commits/5c156aed30404a2a62c0b5d45c80f843e347196a